app/backend/Services/AzureComputerVisionService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace MinimalApi.Services;
+
+public class AzureComputerVisionService(string endPoint, string apiKey)
+{
+    public record ImageEmbeddingResponse(string modelVersion, float[] vector);
+
+    // add virtual keyword to make it mockable
+    public virtual async Task<ImageEmbeddingResponse> VectorizeImageAsync(string imagePathOrUrl, CancellationToken ct = default)
+    {
+        var api = $"{endPoint}/computervision/retrieval:vectorizeImage?api-version=2023-02-01-preview&modelVersion=latest";
+        // first try to read as local file
+        if (File.Exists(imagePathOrUrl))
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Post, api);
+
+            // set authorization header
+            request.Headers.Add("Ocp-Apim-Subscription-Key", apiKey);
+
+            // set body
+            var bytes = await File.ReadAllBytesAsync(imagePathOrUrl, ct);
+            request.Content = new ByteArrayContent(bytes);
+            request.Content.Headers.ContentType = new MediaTypeHeaderValue("image/*");
+
+            // send request
+            using var client = new HttpClient();
```

Comment:
Instead of disposing the `HttpClient` we should be either injecting the `IHttpClientFactory` into this class, and creating a client from that - or at the very least this should be `static` and class-scoped. 

https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory